doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,175 @@
+- Reuse all options used in vic-machine create
+
+  In vic-machine create, we already have 57 options to define VCH configurations, though not all of them can be updated, quite amount of them should be able to be changed, and this number will grow up over time, so we definitely do not want to introduce 3X that number of options to reconfigure (update means add/modify/remove)
+
+## Options
+First option is to reconfigure all in one command
+
+### Implicit Add/Modify and Explicit Delete
+All existing vic-machine create options are supported in vic-machine reconfigure command. The meaning of those options are as following:
+- for elements with labels (e.g. container-network, volume-store), if there's a value provided against that label in the reconfigure, it's a modify operation
+- for elements without labels (e.g. --cert), if there's a value provided it's a replacement
+- if need to remove existing configuration (e.g. static ip configuration for client-network, or registry-ca), define new options as --clear-registry-cas, --clear-client-network. Then existing configuration for registry-ca and/or client-network will be cleared. If not new configuration e.g. for client-network specified, it will share with other networks with same logic in vic-machine create.
+
+The benefit of this option is it can update all things need to be updated in one command, but will introduce a bit more new delete options.
+
+### Separate Reconfigure to Sub Commands
+The idea of this option is to update a few configuration each time, which is one delta update option. For example, to update volume-store, we can use following command
+```
+vic-machine reconfigure volume-store --rm ds://datastore/volume:default
+vic-machine reconfigure volume-store --add nfs://host:port/container:nfs
+```
+
+The command format is like ```vic-machine reconfigure <reconfigure object> --<reconfigure keyword> --<reconfigure option> <configuration value>```
+
+Pros:
+- The whole configuration options are splitted to many sub commands. Then in each sub command, only a small number of options are available, and that does not increase over time. Only the sub command number will increase.
+- Consistent with existing update firewall command format (though the command name might be different)
+
+Cons:
+- We'll need to reinvent option names
+
+  For example, to update client network, user need to specify all options if they want to use static ip. Then in one update command, we'll need all those options. Here is the command looks like ```vic-machine reconfigure bridge-network --port-group value --gateway value --ip value```
+  The existing option name in vic-machine create is ```--client-network value   --client-network-gateway value  --client-network-ip value```, we'll need to remove the prefix cause that is already shown as sub-command name.
+
+Preference:
+
+- Looks like the first option is better, just leave the sub command option here, in case someone could find out more benefit from that option
```

Comment:
It seems to me the clearer and more user friendly approach to this would be to just pass all parameters as you would to vic-machine create but to vic-machine reconfigure? Not adding any new options and using all the same names and formats that create uses. It doesn't look like either approach does that though?  The user should also be able to pass one or more parameters that they want changed?